app/backend/src/clj/teet/contract/contract_db.clj

```diff
@@ -232,6 +232,27 @@
             db contract-id)
        first))
 
+(defn get-lead-partner-company [db contract-id]
+  (let [lead-partner-entities (contract-lead-partner-entities db contract-id)]
+    (when (some? lead-partner-entities)
+          (d/pull db '[*]
+               (first lead-partner-entities)))))
+
+(defn contract-with-lead-partner
+  "Can be used in contract card to show only Lead partner"
+  [db contract]
+  (let [contract-eid (:thk.contract/procurement-id+procurement-part-id contract)
+        lead-partner (get-lead-partner-company db [:thk.contract/procurement-id+procurement-part-id contract-eid])]
+    (assoc contract :thk.contract/lead-partner lead-partner)))
+
+(defn contract-with-all-partners
+  "Used in contract card to show all partners"
+  [db contract]
+  (let [contract-eid (:thk.contract/procurement-id+procurement-part-id contract)
+        partners (get-contract-with-partners db
+                   [:thk.contract/procurement-id+procurement-part-id contract-eid])]
+    (assoc contract :thk.contract/contract-with-partners partners)))
```

Comment:
This fn is passed the whole contract information and in the `get-contract-with-partners` you are pulling all of the contract information again? And associng it to the` contract-with-partners?` key. Should you only fetch the partners or just add the company keys to the pull in the contract listing query. `'[* {:company-contract/_contract [{:company-contract/company [*]}] }]`